### ISSUE TITLE
In Java programs, when a 400 Error is responded to, it is handled as a "CognitoIdentityProviderException"

### ISSUE THREAD

**Issue by HirokiItoga** [jagregory/cognito-local#315]
The problem with the current code is that when a 400 Error is responded to, it is handled as a "CognitoIdentityProviderException" in all Java programs (For example, if you request with an auth_code different from the auth_code generated by signup, we want a CodeMismatchException to be thrown).

https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/cognitoidentityprovider/model/CognitoIdentityProviderException.html
https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/cognitoidentityprovider/model/CodeMismatchException.html

This is because the latest AWS SDK for Java JsonErrorCodeParser identifies the Class of Exception by referencing the value of the Response json key as "__type"(JsonErrorCodeParser is determining the Exception).
(Latest version of AWS SDK for Java is 2.17 (2022/09/09))
https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/protocols/json/internal/unmarshall/JsonErrorCodeParser.html

We know how to fix this problem so we created below PR.
#304

**Comment by paullallier** [comment]
It looks like this also applies to the AWS SDK for PHP.  For example, adminGetUser - user not found looks like this from Cognito-proper:
```
Error executing "AdminGetUser" on "https://cognito-idp.eu-west-2.amazonaws.com"; AWS HTTP error: Client error: `POST https://cognito-idp.eu-west-2.amazonaws.com` resulted in a `400 Bad Request` response:
{"__type":"UserNotFoundException","message":"User does not exist."}
UserNotFoundException (client): User does not exist. - {"__type":"UserNotFoundException","message":"User does not exist."}
```

And this in cognito-local (note, I'm spoofing the DNS to use an almost-identical server name in my test environment):
```
Error executing "AdminGetUser" on "https://cognito-idp.eu-west-2-test.amazonaws.com"; AWS HTTP error: Client error: `POST https://cognito-idp.eu-west-2-test.amazonaws.com` resulted in a `400 Bad Request` response:
{"code":"CognitoLocal#UserNotFoundException","message":"User does not exist"}
  (client):  - {"code":"CognitoLocal#UserNotFoundException","message":"User does not exist"}

```

**Comment by jagregory** [comment]
This may have been fixed in [v3.21.2](https://github.com/jagregory/cognito-local/releases/tag/v3.21.2). Please test and re-open this issue if the problem isn't fixed.
